group-generators/generators/lilnouns-proplot-contributors/index.ts

```diff
@@ -0,0 +1,67 @@
+import { dataProviders } from "@group-generators/helpers/providers";
+import { ApiConfig } from '@group-generators/helpers/providers/rest-api';
+import { Tags, ValueType, GroupWithData } from "topics/group";
+import {
+  GenerationContext,
+  GenerationFrequency,
+  GroupGenerator,
+} from "topics/group-generator";
+
+type ApiDataType = {
+  id: number;
+  createdAt: string;
+  updatedAt: string;
+  title: string;
+  tldr: string;
+  description: string;
+  creatorId: string;
+  votecount: number;
+};
+
+type FilteredDataType = {
+  creatorId: string;
+  votecount: number;
+};
+
+const generator: GroupGenerator = {
+  generationFrequency: GenerationFrequency.Once,
+
+  generate: async (context: GenerationContext): Promise<GroupWithData[]> => {
+    const ApiConfig: ApiConfig = {
```

Comment:
How are you defining the requirements for the Top Contributor badge? It looks like you're just checking if they have posted an idea with a votecount >= 50? Does that === top contributor?

I'd have thought looking at a combination of the metrics in the userStats property on the `getAllUsers` query may be a better indicator of contributions? You have their vote, comment and idea counts available on the query. 

```
type UserStats {
  totalVotes: Int
  totalComments: Int
  totalIdeas: Int
}
```

Even downvoted ideas are valuable imo so i wouldn't exclude them from being a "Top Contributor" but yeah comes back to how you want to define that.

Reply:
Hey, @ThatTobMate thanks for the feedback but you see there are 2 types of top participants in this PR:
- one who has published ideas, having a votecount > **50**
- and the other one is top voters, having voted on at least **15** ideas (please refer [this](https://github.com/sismo-core/sismo-hub/pull/156/files#diff-50a0fa0e987247f3528d3c1ae7d50f0ef224a52b721348c1657e2a1d5b79171cR26-R31), maybe you missed this one)

And yes, I did think about top commenters + frequent ideas contributors, but the reason I've not included them here is, I wanted to start small and gradually add more metrics as we move forward. And the other reason is how Sismo stores these values. Read the below note.

> **Sismo** stores these values in a Merkle tree, in the form of a **`key`: `value`** pair where **`key`** is the address & the **`value`** could be anything for example: **`votecount`**, **`totalVotes`**, **`totalComments`**. You can't store an array here. So if you want to have all these metrics, you would have to design a **weighted scoring model** to generate a score that you can then assign to the `value` of its corresponding `key`.

But I thought it will be too much overengineered considering the current statistics of PropLot. We can definitely do this in the next iterations. 

Let me know if this answers your concern. 🙂